sophiread/SophireadCLI/CMakeLists.txt

```diff
@@ -30,22 +51,61 @@ add_executable(
     tests/test_user_config.cpp
     src/user_config.cpp
 )
-target_link_libraries(
-    UserConfigTest
-    FastSophiread
-    spdlog::spdlog
-    GTest::GTest
-    GTest::Main
+target_link_libraries(UserConfigTest
+    PRIVATE 
+        FastSophiread
+        spdlog::spdlog
+        GTest::GTest
+        GTest::Main
 )
 gtest_discover_tests(UserConfigTest)
+# Json config test
+add_executable(
+    JsonConfigParserTest
+    tests/test_json_config_parser.cpp
+    src/json_config_parser.cpp
+)
+target_link_libraries(JsonConfigParserTest
+    PRIVATE 
+        FastSophiread
+        spdlog::spdlog
+        GTest::GTest
+        GTest::Main
+        nlohmann_json::nlohmann_json
+)
+gtest_discover_tests(JsonConfigParserTest)
+# core test
+add_executable(
+    SophireadCoreTest
+    tests/test_sophiread_core.cpp
+    src/sophiread_core.cpp
+    src/json_config_parser.cpp
+)
+target_link_libraries(SophireadCoreTest
+    PRIVATE 
+        FastSophiread
+        spdlog::spdlog
+        GTest::GTest
+        GTest::Main
+        nlohmann_json::nlohmann_json
+        ${HDF5_LIBRARIES}
+        ${TIFF_LIBRARIES}
+        TBB::tbb
+)
+gtest_discover_tests(SophireadCoreTest)
 
 # ----------------- SYMLINK ----------------- #
 # symlink the executable to the build directory
 add_custom_command(TARGET Sophiread POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E create_symlink 
+    COMMAND ${CMAKE_COMMAND} -E create_symlink
     ${PROJECT_BINARY_DIR}/SophireadCLI/Sophiread
     ${PROJECT_BINARY_DIR}/Sophiread
 )
+add_custom_command(TARGET venus_auto_reducer POST_BUILD
```

Comment:
I've not seen a symlink created by cmake before. I assume it creates a symlink, but why do you need to?

Reply:
I link it to the root of build to make it easier for me to test them